tests/algorithms/test_factorize_algorithm.py

```diff
@@ -48,41 +53,78 @@ def test_factorize_surgery(state: State, empty_logger: Logger, algo_instance: Fa
 
     if algo_instance.factorize_convs:
         num_factorized_layers = module_surgery.count_module_instances(state.model, FactorizedConv2d)
-        assert num_conv_layers == num_factorized_layers
```

Comment:
Could you add a comment explaining the new assert? It's not obvious to me what is doing on here

Reply:
I've added this comment: "Each Conv2d/Linear is either unmodified or replaced with a factorized version, which contains two Conv2d/Linears submodules."

The assertion just checks that this holds.